logdb/logdb.go

```diff
@@ -72,8 +74,75 @@ func New(path string) (logDB *LogDB, err error) {
 }
 
 // NewMem create a log db in ram.
+// Unlike the file-based database, this implementation:
+// 1. Uses synchronous=off for faster writes in memory
+// 2. Uses a single connection with minimal transaction overhead
+// 3. Optimizes for in-memory performance
 func NewMem() (*LogDB, error) {
-	return New("file::memory:")
+	// Generate a unique database name to prevent concurrent access issues
+	dbName := fmt.Sprintf("file:memdb_%d?mode=memory&cache=shared&_txlock=immediate&synchronous=off&journal_mode=memory", time.Now().UnixNano())
+
+	// Open SQLite in-memory database with optimized settings:
+	// - mode=memory: Create a new in-memory database
+	// - cache=shared: Enable shared cache mode
+	// - _txlock=immediate: Acquire locks immediately for better concurrency
+	// - synchronous=off: Fastest mode for in-memory database
+	// - journal_mode=memory: Use in-memory journal for better performance
+	db, err := sql.Open("sqlite3", dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set optimized PRAGMA settings for in-memory performance
+	if _, err := db.Exec(`
+		PRAGMA synchronous = OFF;
```

Comment:
`synchronous` and `journal_mode` already declared in the connection string, duplicated here?